src/Centreon/Application/Controller/PlatformTopologyController.php

```diff
@@ -0,0 +1,130 @@
+use Centreon\Domain\PlatformTopology\Interfaces\PlatformTopologyServiceInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * This controller is designed to manage platform topology API requests and register new servers.
+ *
+ * @package Centreon\Application\Controller
+ */
+class PlatformTopologyController extends AbstractController
+{
+    /**
+     * @var PlatformTopologyServiceInterface
+     */
+    private $platformTopologyService;
+
+    /**
+     * PlatformTopologyController constructor
+     * @param PlatformTopologyServiceInterface $platformTopologyService
+     */
+    public function __construct(PlatformTopologyServiceInterface $platformTopologyService)
+    {
+        $this->platformTopologyService = $platformTopologyService;
+    }
+
+    /**
+     * Validate platform topology data according to json schema
+     *
+     * @param array<mixed> $platformToAdd data sent in json
+     * @param string $schemaPath
+     * @return void
+     * @throws PlatformTopologyException
+     */
+    private function validatePlatformTopologySchema(array $platformToAdd, string $schemaPath): void
+    {
+        $platformTopologySchemaToValidate = Validator::arrayToObjectRecursive($platformToAdd);
+        $validator = new Validator();
+        $validator->validate(
+            $platformTopologySchemaToValidate,
+            (object) ['ref' => 'file://' . $schemaPath],
+            Constraint::CHECK_MODE_VALIDATE_SCHEMA
+        );
+
+        if (!$validator->isValid()) {
+            $message = '';
+            foreach ($validator->getErrors() as $error) {
+                $message .= sprintf("[%s] %s\n", $error['property'], $error['message']);
+            }
+            throw new PlatformTopologyException($message);
+        }
+    }
+
+    /**
+     * Entry point to register a new server
+     *
+     * @param Request $request
+     * @return View
+     * @throws PlatformTopologyException
+     */
+    public function addPlatformToTopology(Request $request): View
+    {
+        // check user rights
+        $this->denyAccessUnlessGrantedForApiConfiguration();
+
+        // get http request content
+        $platformToAdd = json_decode((string) $request->getContent(), true);
+        if (!is_array($platformToAdd)) {
+            throw new PlatformTopologyException(
+                _('Error when decoding sent data'),
+                Response::HTTP_BAD_REQUEST
+            );
+        }
+
+        // validate data consistency
+        $this->validatePlatformTopologySchema(
+            $platformToAdd,
+            $this->getParameter('centreon_path')
+            . 'config/json_validator/latest/Centreon/PlatformTopology/Register.json'
+        );
+
+        try {
+            $platformTopology = (new PlatformTopology())
+                ->setName($platformToAdd['name'])
+                ->setType($platformToAdd['type'])
+                ->setAddress($platformToAdd['address']);
```

Comment:
```suggestion
                ->setAddress($platformToAdd['address'])
                ->setType($platformToAdd['type']);
```
"address" needs to be checked and set before "type", to be able to throw correctly the message
![image](https://user-images.githubusercontent.com/34628915/89525560-ab9f1180-d7e6-11ea-84e5-8573ba2885d4.png)